containers/ome-tiff-offsets/context/main.py

```diff
@@ -2,26 +2,35 @@
 from glob import glob
 from pathlib import Path
 from os import makedirs
+from os.path import splitext
 import json
+from os.path import dirname
+from os.path import exists
+from os import makedirs
 
 from skimage.external import tifffile
 
-
 def get_offsets(tiff_filepath):
     with tifffile.TiffFile(tiff_filepath) as tif:
         offsets = [page._offset for page in tif.pages]
     return offsets
 
 def main(input_dir, output_dir):
     makedirs(output_dir, exist_ok=True)
-    for input_path in glob(input_dir + '/*.ome.tiff') + glob(input_dir + '/*.ome.tif'):
+
+    # Find all OME.TIFFs in the input directory.
+    for input_path in sum([Path(input_dir).glob('**/*.ome.{ext}') for ext in ['tif', 'tiff'], []]):
```

Comment:
Is this even valid syntax? It doesn't seem like it:

```python
Python 3.8.2 (default, Apr 27 2020, 15:53:34) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.16.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: def get_offsets(*args): 
   ...:     pass 
   ...:                                                                                                                                                     

In [2]: from pathlib import Path                                                                                                                            

In [3]: input_dir = Path()                                                                                                                                  

In [4]: for input_path in sum([Path(input_dir).glob('**/*.ome.{ext}') for ext in ['tif', 'tiff'], []]): 
   ...:     offsets = get_offsets(input_path)                                                                                                               
  File "<ipython-input-4-4a3156397fbc>", line 1
    for input_path in sum([Path(input_dir).glob('**/*.ome.{ext}') for ext in ['tif', 'tiff'], []]):
                                                                                            ^
SyntaxError: invalid syntax
```

In any case, the usage of `sum` here looks a little odd -- perhaps `itertools.chain` or encapsulating the "find all OME-TIFF files with whatever extension" functionality into a separate function like in https://github.com/hubmapconsortium/ome-tiff-pyramid/blob/master/bin/collect_ometiff_files.py#L12 ?

Reply:
@mruffalo since your ome-tiff-pyramid only produces `tif` files the code is not longer looking for `tiff`. Since offsets are supposed to be computed from the pyramid file.